### DISCUSSION TITLE
Invalid access of closed uWS.WebSocket/SSLWebSocket.

### DISCUSSION THREAD

**Discussion by davidshore** [uNetworking/uWebSockets.js#508] [Q&A] [answered]
Here I have a loop where my game server is sending game invites to some players online depending on their elo score:
Each roomPlayer object has a webSocket (ws) object as a property.

```
for (const username of usernamesInRoom) {
    const roomPlayer = players[username];
      const elo = roomPlayer.getElo(); 
      if (elo >= match.minElo && elo <= match.maxElo) {
        try {
          roomPlayer.ws.send(quickMatchStr);
        } catch (e) {
          console.warn('quick match', e);
          continue;
        }
      }
  }
```

I keep getting these errors: quick match Invalid access of closed uWS.WebSocket/SSLWebSocket.

How can I check that the websocket is not closed before I try to send?

**Comment by hst-m** [comment, marked as answer]
In the websocket close event handler you can mark the ws as closed or remove player from list `close:ws=>ws.closed=true`
